### PR TITLE
Add tests for organisation schema test cases

### PR DESCRIPTION
**Pull request by dalepotter**
This adds the `iati-organisations-schema.xsd` file to the testrunner.

Code requires refactoring to make more DRY.

### CODE REVIEW

File: tests/test_schemas.py [hunk 1]
```diff
@@ -54,27 +68,42 @@ def load_as_dataset(filepath):
     return iati.Dataset(xml_str)
 
 
-@pytest.mark.parametrize('filepath', list_xml_files_recursively('tests/activity-tests/should-pass/') +  # Legacy tests
-                         list_xml_files_recursively('tests/should-pass/'))  # Tests in new format
-def test_pass_files(activity_schema, filepath):
-    """Check that all 'should-pass' test files are XML and pass Schema validation."""
+@pytest.mark.parametrize('filepath', list_xml_files_recursively('tests/activity-tests/should-pass/') +  # Legacy activity test cases
+                         list_xml_files_recursively('tests/should-pass/iati-activities/') +  # Legacy activity test cases
```
Comment (allthatilk) [inline tests/test_schemas.py]: Is this coment correct? I thought tests/should-pass were the new test cases for 2.03?

File: tests/test_schemas.py [hunk 2]
```diff
@@ -54,27 +68,42 @@ def load_as_dataset(filepath):
...
+                         list_xml_files_recursively('tests/organisation-tests/should-pass/') +  # Legacy organisaion test cases
+                         list_xml_files_recursively('tests/should-pass/iati-organisations/'))  # Organisation test cases in new format
+def test_pass_files(activity_schema, organisation_schema, filepath):
+    """Check that all activity and organisation schema 'should-pass' test files are XML and pass Schema validation."""
     dataset = load_as_dataset(filepath)
 
+    if 'activity-tests' in filepath or 'iati-activities' in filepath:
+        schema = activity_schema
+    else:
```
Comment (allthatilk) [inline tests/test_schemas.py]: Not sure the `else` is doing much here, could it be implicit?

File: tests/test_schemas.py [hunk 3]
```diff
@@ -54,27 +68,42 @@ def load_as_dataset(filepath):
...
+
+@pytest.mark.parametrize('filepath', list_xml_files_recursively('tests/activity-tests/should-fail/') +
+                         list_xml_files_recursively('tests/organisation-tests/should-fail/'))
+def test_fail_files(activity_schema, organisation_schema, filepath):
+    """Check that all legacy activity and organisation 'should-fail' test files are XML but fail Schema validation."""
     dataset = load_as_dataset(filepath)
 
+    if 'activity-tests' in filepath:
+        schema = activity_schema
+    else:
```
Comment (allthatilk) [inline tests/test_schemas.py]: As above

Comment (allthatilk) [reply tests/test_schemas.py]: Though might want to extract this as schema detection generally anyway.

File: tests/test_schemas.py [hunk 4]
```diff
@@ -87,7 +116,12 @@ def test_2_03_fail_files(activity_schema, filepath):
     filename_no_extension = filename.split('.')[0]
     failure_reason = filename_no_extension.split("_")[0]
 
-    error_log = iati.validator.validate_is_iati_xml(dataset, activity_schema)
+    if 'iati-activities' in filepath:
```
Comment (allthatilk) [inline tests/test_schemas.py]: I'm guessing this is one of the non-DRY things. Think it's worth extracting it.

**Review by allthatilk — CHANGES_REQUESTED** [review]
Just a few changes. Think it is worth extracting out schema detection.

File: tests/test_schemas.py (hunk 2 above)
Comment (dalepotter) [reply tests/test_schemas.py]: Changed to an elif in as explicit is better than implicit in fda3cb9.

File: tests/test_schemas.py (hunk 1 above)
Comment (dalepotter) [reply tests/test_schemas.py]: Fixed in 684e12b.

File: tests/test_schemas.py (hunk 3 above)
Comment (dalepotter) [reply tests/test_schemas.py]: Fixed in a86cf85.

File: tests/test_schemas.py (hunk 4 above)
Comment (dalepotter) [reply tests/test_schemas.py]: Fixed in a86cf85.

File: tests/should-fail/iati-organisations/iati-organisation/err-not-iati-xml-missing-required-element_no-required-elements-exist.xml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0"?>
+
+<iati-organisations version="xx">
+
+  <iati-organisation>
+    <!-- No required elements exist as children of iati-organisation-->
```
Comment (allthatilk) [inline tests/should-fail/iati-organisations/iati-organisation/err-not-iati-xml-missing-required-element_no-required-elements-exist.xml]: Need to add `PASS: ` to start of comment for consistency across 2.03 test data.

File: tests/test_schemas.py [hunk 6]
```diff
@@ -54,40 +87,47 @@ def load_as_dataset(filepath):
     return iati.Dataset(xml_str)
 
 
-@pytest.mark.parametrize('filepath', list_xml_files_recursively('tests/activity-tests/should-pass/') +  # Legacy tests
-                         list_xml_files_recursively('tests/should-pass/'))  # Tests in new format
-def test_pass_files(activity_schema, filepath):
-    """Check that all 'should-pass' test files are XML and pass Schema validation."""
+@pytest.mark.parametrize('filepath', list_xml_files_recursively('tests/activity-tests/should-pass/') +  # Legacy activity test cases
+                         list_xml_files_recursively('tests/should-pass/iati-activities/') +  # Activity test cases in new format
```
Comment (allthatilk) [inline tests/test_schemas.py]: new format -> `2.03+ format`

File: tests/test_schemas.py [hunk 7]
```diff
@@ -54,40 +87,47 @@ def load_as_dataset(filepath):
...
+                         list_xml_files_recursively('tests/organisation-tests/should-pass/') +  # Legacy organisaion test cases
+                         list_xml_files_recursively('tests/should-pass/iati-organisations/'))  # Organisation test cases in new format
```
Comment (allthatilk) [inline tests/test_schemas.py]: as above

File: tests/test_schemas.py [hunk 8]
```diff
@@ -54,40 +87,47 @@ def load_as_dataset(filepath):
...
+def test_pass_files(schema_factory, filepath):
+    """Check that all activity and organisation schema 'should-pass' test files are XML and pass Schema validation."""
     dataset = load_as_dataset(filepath)
+    schema = schema_factory.get_correct_schema_for_filepath(filepath)
 
     assert iati.validator.is_xml(dataset)
-    assert iati.validator.is_iati_xml(dataset, activity_schema)
+    assert iati.validator.is_iati_xml(dataset, schema)
+
```
Comment (allthatilk) [inline tests/test_schemas.py:101]: Should this be 2 blank lines? Not sure but it looks odd...

**Review by allthatilk — CHANGES_REQUESTED** [review]
Some very small things and then it's good to go!

File: tests/test_schemas.py (hunk 6 above)
Comment (dalepotter) [reply tests/test_schemas.py]: Fixed in 54599ae.

File: tests/test_schemas.py (hunk 7 above)
Comment (dalepotter) [reply tests/test_schemas.py]: Fixed in 54599ae.

File: tests/should-fail/iati-organisations/iati-organisation/err-not-iati-xml-missing-required-element_no-required-elements-exist.xml (hunk 5 above)
Comment (dalepotter) [reply tests/should-fail/iati-organisations/iati-organisation/err-not-iati-xml-missing-required-element_no-required-elements-exist.xml]: Fixed in 303e54f

File: tests/test_schemas.py (hunk 8 above)
Comment (dalepotter) [reply tests/test_schemas.py:101]: According to the linter, there should be two blank lines between function definitions.

Comment (allthatilk) [reply tests/test_schemas.py:101]: Ah, there is now. Good good!

**Review by allthatilk — APPROVED** [review]
Looks good!

**Pull request merged**